### PR TITLE
IGNITE-21916 Sql. Cover SQL T828 (JSON_QUERY) feature by tests

### PR DESCRIPTION
**Pull request by xtern**
https://issues.apache.org/jira/browse/IGNITE-21916

### CODE REVIEW

File: modules/sql-engine/src/integrationTest/java/org/apache/ignite/internal/sql/engine/ItSqlOperatorsTest.java [hunk 1]
```diff
@@ -326,6 +331,25 @@ public void testJson() {
         assertExpression("'{\"a\":1}' IS NOT JSON OBJECT").returns(false).check();
         assertExpression("'[1, 2]' IS NOT JSON ARRAY").returns(false).check();
         assertExpression("'1' IS NOT JSON SCALAR").returns(false).check();
+
+        // TODO https://issues.apache.org/jira/browse/IGNITE-20163
+        // assertExpression("'{\"a\":1}' FORMAT JSON").check();
+    }
+
+    @Test
+    @SuppressWarnings("ThrowableNotThrown")
```
Comment (ygerzhedovich) [inline modules/sql-engine/src/integrationTest/java/org/apache/ignite/internal/sql/engine/ItSqlOperatorsTest.java:340]: it looks like our gap in support of JSON. I suggest to create a ticket to support it, rewrite test  and mute it under the newly created ticket

File: modules/sql-engine/src/integrationTest/sql/function/json/test_json.test [hunk 2]
```diff
@@ -26,3 +26,28 @@ query R
...
+{"age":32,"Name":"Alex"}
+{"age":27,"Name":"Maria"}
+
+query T
+select json_array(name, age) from t
+----
+["Alex",32]
+["Maria",27]
+
+# Note nested object construction fails with NullPointerException
```
Comment (ygerzhedovich) [inline modules/sql-engine/src/integrationTest/sql/function/json/test_json.test:48]: it should be muted with JIRA ticket

File: modules/sql-engine/src/integrationTest/java/org/apache/ignite/internal/sql/engine/ItSqlOperatorsTest.java (hunk 1 above)
Comment (xtern) [reply modules/sql-engine/src/integrationTest/java/org/apache/ignite/internal/sql/engine/ItSqlOperatorsTest.java]: Fixed, thanks

File: modules/sql-engine/src/integrationTest/sql/function/json/test_json.test (hunk 2 above)
Comment (xtern) [reply modules/sql-engine/src/integrationTest/sql/function/json/test_json.test]: Fixed, thanks

**Review by ygerzhedovich — APPROVED** [review]

**Pull request merged**